pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -235,6 +253,9 @@ func (w *Watcher) handleCreateEvent(event fsnotify.Event) error {
 		return w.handlePluginRegistration(event.Name)
 	}
 
+	if w.containsBlacklistedDir(event.Name) {
```

Comment:
is there a reason we don't do this check at the top of the function? Are there types of updates that would exercise some of the code paths above and give spurious errors/warnings (stat errors, non-dir socket files that would trigger calls to handlePluginRegistration incorrectly, etc)?

Reply:
Good point moved.